master/internal/api_tensorboard.go

```diff
@@ -80,20 +103,9 @@ func (a *apiServer) GetTensorboards(
 		return nil, err
 	}
 
-	a.filter(&resp.Tensorboards, func(i int) bool {
-		if err != nil {
-			return false
-		}
-		ok, serverError := command.AuthZProvider.Get().CanGetNSC(
-			ctx, *curUser, model.UserID(resp.Tensorboards[i].UserId), command.PlaceHolderWorkspace)
-		if serverError != nil {
-			err = serverError
-		}
-		return ok
-	})
-	if err != nil {
-		return nil, err
-	}
+	filteredTensorboards, err := a.filterTensorboards(ctx, *curUser, resp.Tensorboards,
```

Comment:
do we need to check this error? 
question: also would this return a 404 if the user requests workspace that exists but they don't have access to?

Reply:
doh good catch, was trying to be quick but was sloppy. The basic implementation will not restrict the user in any way, so it should never return any error. For EE, it depends on how we implement the function. I think we should just return an empty list if they don't have access to a workspace.

Reply:
related conversation https://determined-ai.slack.com/archives/C03LE5Z8R46/p1672776613155459?thread_ts=1672772755.830529&cid=C03LE5Z8R46 I think the idea there is that we probably want to keep it similar to asking for a workspace that doesn't exist (we can handle this situation in a separate pr)